internal/services/appservice/windows_function_app_slot_resource.go

```diff
@@ -0,0 +1,778 @@
+			}
+			if err := updateFuture.WaitForCompletionRef(ctx, client.Client); err != nil {
+				return fmt.Errorf("waiting to update %s: %+v", id, err)
+			}
+
+			if _, err := client.UpdateConfigurationSlot(ctx, id.ResourceGroup, id.SiteName, web.SiteConfigResource{SiteConfig: siteConfig}, id.SlotName); err != nil {
+				return fmt.Errorf("updating Site Config for Windows %s: %+v", id, err)
+			}
+
+			if metadata.ResourceData.HasChange("connection_string") {
+				connectionStringUpdate := helpers.ExpandConnectionStrings(state.ConnectionStrings)
+				if connectionStringUpdate.Properties == nil {
+					connectionStringUpdate.Properties = map[string]*web.ConnStringValueTypePair{}
+				}
+				if _, err := client.UpdateConnectionStringsSlot(ctx, id.ResourceGroup, id.SiteName, *connectionStringUpdate, id.SlotName); err != nil {
+					return fmt.Errorf("updating Connection Strings for Windows %s: %+v", id, err)
+				}
+			}
+
+			if metadata.ResourceData.HasChange("auth_settings") {
+				authUpdate := helpers.ExpandAuthSettings(state.AuthSettings)
+				if _, err := client.UpdateAuthSettingsSlot(ctx, id.ResourceGroup, id.SiteName, *authUpdate, id.SlotName); err != nil {
+					return fmt.Errorf("updating Auth Settings for Windows %s: %+v", id, err)
+				}
+			}
+
+			if metadata.ResourceData.HasChange("backup") {
+				backupUpdate := helpers.ExpandBackupConfig(state.Backup)
+				if backupUpdate.BackupRequestProperties == nil {
+					if _, err := client.DeleteBackupConfigurationSlot(ctx, id.ResourceGroup, id.SiteName, id.SlotName); err != nil {
+						return fmt.Errorf("removing Backup Settings for Windows %s: %+v", id, err)
+					}
+				} else {
+					if _, err := client.UpdateBackupConfigurationSlot(ctx, id.ResourceGroup, id.SiteName, *backupUpdate, id.SlotName); err != nil {
+						return fmt.Errorf("updating Backup Settings for Windows %s: %+v", id, err)
+					}
+				}
+			}
+
+			if metadata.ResourceData.HasChange("site_config.0.app_service_logs") {
+				appServiceLogs := helpers.ExpandFunctionAppAppServiceLogs(state.SiteConfig[0].AppServiceLogs)
+				if _, err := client.UpdateDiagnosticLogsConfigSlot(ctx, id.ResourceGroup, id.SiteName, appServiceLogs, id.SlotName); err != nil {
+					return fmt.Errorf("updating App Service Log Settings for %s: %+v", id, err)
+				}
+			}
+
+			return nil
+		},
+	}
+}
+
+func (m *WindowsFunctionAppSlotModel) unpackWindowsFunctionAppSettings(input web.StringDictionary) {
+	if input.Properties == nil {
+		return
+	}
+
+	appSettings := make(map[string]string)
+	var dockerSettings helpers.ApplicationStackDocker
+	m.BuiltinLogging = false
+
+	for k, v := range input.Properties {
+		switch k {
+		case "FUNCTIONS_EXTENSION_VERSION":
+			m.FunctionExtensionsVersion = utils.NormalizeNilableString(v)
+
+		case "WEBSITE_NODE_DEFAULT_VERSION": // Note - This is only set if it's not the default of 12, but we collect it from WindowsFxVersion so can discard it here
+		case "WEBSITE_CONTENTAZUREFILECONNECTIONSTRING":
```

Comment:
Do we need to do anything for these next 4 cases?

Reply:
As above...